.github/workflows/ubuntu.yml

```diff
@@ -150,23 +148,22 @@ jobs:
           make -j8
           mv validate ../
           cd ../
-          ls -la
           rm -rf VAL
 
       - name: archive-files
         # We only run tests on the version compiled with gcc, so we
         # only need to archive that one.
-        if: ${{ matrix.compiler-version }} == gcc
+        if: ${{ matrix.compiler-version.cc == 'gcc' }}
         run: |
           cd ../
-          tar cfz ${{ matrix.compiler-version }}.tar.gz downward lib
+          tar cfz ${{ matrix.compiler-version.cc }}.tar.gz downward lib
 
       - name: upload-files
-        if: ${{ matrix.compiler-version }} == gcc
+        if: ${{ matrix.compiler-version.cc == 'gcc' }}
         uses: actions/upload-artifact@v1
         with:
           name: compiled-planner-${{ matrix.ubuntu-version }}
-          path: /home/runner/work/downward/${{ matrix.compiler-version }}.tar.gz
+          path: /home/runner/work/downward/${{ matrix.compiler-version.cc }}.tar.gz
```

Comment:
I just pushed the current main branch to a private repo and even though only the macOS and Windows build steps succeeded, 2.1 GB of artifacts were uploaded. These artifacts are actually retained for 90 days and the limit is 2 GB. Not sure where the 2.1 GB come from. The debug and release Windows tarballs come in at 200 MB and the macOS tarballs are both 50 MB in size. In any case, we have to delete the artifacts after a job is done, I think. (That's not a solution for jobs with failing tests though.) Or we try to find a solution that doesn't store artifacts in the first place.

Reply:
Using a private repo to run these actions is not a good idea, because also the runtime limit will be reached after a few runs at last. I don't think this is a problem for us.

With the ubuntu artifact, we could certainly only upload include the driver instead of the whole code, but I doubt this will reduce size a lot. We upload the entire installation of external libraries which probably consume a lot of space. It seems to me that deleting artifacts is the right solution.

Reply:
> Using a private repo to run these actions is not a good idea, because also the runtime limit will be reached after a few runs at last. I don't think this is a problem for us.

I agree. This is the default, however. At the moment users would have to disable GitHub Actions for all private repos manually.

Regarding artifacts, I see several TODOs:
1. We should set the retention time to the minimum: 1 day. This is done by adding `retention-days: 1` to the upload step (https://github.blog/changelog/2020-10-08-github-actions-ability-to-change-retention-days-for-artifacts-and-logs/).
2. We should delete artifacts after the tests finish (ideally for failed tests as well).
3. We should try to make the tarballs smaller. Currently they are so large (> 700 MB sometimes) that uploading them takes 30s-60s. At the bottom of this page (https://github.com/aibasel/downward/actions/runs/558175941) you can see the artifact sizes for Windows builds.

I wonder whether the whole artifact dance is really needed. Can't we just use an if-statement to only run the tests for a subset of the matrix combinations?

Reply:
I have not really looked into this but from dealing with the IPC containers I remember that the size of the object files has a large impact. So calling `./build.py` and then zipping `builds` created a much larger package than just zipping `builds/release/bin`.

Reply:
> I wonder whether the whole artifact dance is really needed. Can't we just use an if-statement to only run the tests for a subset of the matrix combinations?

We use them for testing different python versions without installing and compiling everything just for testing several python versions. That is, we upload one compiled version and use it with several tests. We can get rid of this if we are fine with running the same setup steps for testing different pythons. Or maybe we don't need to test running the planner with all python versions?

Reply:
About Silvans last point, do we need to test different versions. My thoughts are:
We have to test the last supported python version. Python might introduce a nice python feature, I get used to it. Then use this feature in the FD driver and we lose support for python3.6. Thus, we have to test our last supported python version. 
If the python developer do not change the behaviour of a function for different python 3.X releases, anything that works for 3.6 should also work for newer versions (unless there is a bug introduced or fixed in python).

Does the behaviour of functions change or is there another reason why code that works for 3.6 should not work anymore for >3.6

Reply:
Yes, we need to test the driver using different Python versions. What I meant was to move all steps into one job, add Python 3.6 and 3.8 to the matrix, and add `if python_version == 3.6` to all compilation steps and `if cc == gcc and cxx == g++` to all test steps (haven't checked the if-syntax).